blog/2021-08-12-state_class_total.md

```diff
@@ -0,0 +1,74 @@
+---
+author: Erik Montnémery
+authorURL: https://github.com/emontnemery
+title: "New sensor state classes: total and total_increasing"
+---
+
+Two new state classes, `STATE_CLASS_TOTAL` and `STATE_CLASS_TOTAL_INCREASING` have been
+added. In addition, it is no longer mandatory to set the `last_reset` attribute in order
+for `sum` statistics to be generated. The driver for the changes is to make it easier to
+integrate with utility meters etc.
+
+### State classes
+
+There are now 3 defined state classes:
+- `STATE_CLASS_MEASUREMENT`, the state represents a measurement in present time, for 
+   example a temperature, electric power etc.
+- `STATE_CLASS_TOTAL`, the state represents a total amount which can both increase and
+   decrease, e.g. the value of a stock portfolio
+- `STATE_CLASS_TOTAL_INCREASING`, a monotonically increasing total, e.g. an amount of
+   consumed gas, water or energy
+
+#### STATE_CLASS_TOTAL
+If the sensor's state class is `STATE_CLASS_TOTAL`, the last_reset attribute can
+optionally be set to gain manual control of meter cycles; each time last_reset changes
+the corresponding value is used as the zero-point when calculating `sum` statistics.
+If last_reset is not set, the sensor's value when it was first added is used as the
+zero-point when calculating `sum` statistics.
+
+Example of `STATE_CLASS_TOTAL` without last_reset:
+
+| t    | state  | sum    |
+| ---: | -----: | -----: |
+|   0  |  1000  |     0  |
+|   1  |  1010  |    10  |
+|   2  |     0  | -1000  |
+|   3  |     5  |  -995  |
+
+Example of `STATE_CLASS_TOTAL` with last_reset:
+
+| t    | state  | last_reset          | sum    |
+| ---: | -----: | ------------------- | -----: |
+|   0  |  1000  | 2021-08-01T13:30:00 |     0  |
+|   1  |  1010  | 2021-08-01T13:30:00 |    10  |
+|   2  |  1005  | 2021-08-01T13:30:00 |     5  |
+|   3  |     0  | 2021-09-01T13:30:00 |     5  |
+|   4  |     5  | 2021-09-01T13:30:00 |    10  |
```

Comment:
The time stamps are for the `last_reset` time though, so they should not continuously increment.
The intention was that the `t` column is time, but we can make that incrementing time stamps instead of just a sequence   of integers.